main/wallet-backend.js

```diff
@@ -182,98 +173,15 @@ class WalletBackend {
     }
   }
 
-  /**
-   * @param {string} cid
-   * @returns {Promise<GQLReceipt>}
-   */
-  async getReceipt (cid) {
-    const query = gql`
-      query Receipt($cid: String!) {
-        receipt(cid: $cid) {
-          return
-          exitCode
-          gasUsed
-        }
-      }
-    `
-    const variables = { cid }
-    /** @type {{receipt: GQLReceipt}} */
-    const { receipt } = await request(this.url, query, variables)
-    return receipt
-  }
-
-  /**
-   * @param {string} cid
-   * @returns {Promise<GQLStateReplay>}
-   */
-  async getStateReplay (cid) {
-    const query = gql`
-      query StateReplay($cid: String!) {
-        stateReplay(cid: $cid) {
-          receipt {
-            return
-            exitCode
-            gasUsed
-          }
-          executionTrace {
-            executionTrace
-          }
-        }
-      }
-    `
-    const variables = { cid }
-    /** @type {{stateReplay: GQLStateReplay}} */
-    const { stateReplay } = await request(this.url, query, variables)
-    return stateReplay
-  }
-
-  /**
-   * @param {number} height
-   * @returns Promise<GQLTipset>
-   */
-  async getTipset (height) {
-    const query = gql`
-      query Tipset($height: Uint64!) {
-        tipset(height: $height) {
-          minTimestamp
-        }
-      }
-    `
-    const variables = { height }
-    const { tipset } = await request(this.url, query, variables)
-    return tipset
-  }
-
   /**
    * @param {string} address
    * @returns Promise<GQLMessage[]>
    */
   async getMessages (address) {
-    const query = gql`
-      query Messages($address: String!, $limit: Int!, $offset: Int!) {
-        messages(address: $address, limit: $limit, offset: $offset) {
-          cid
-          to {
-            robust
-          }
-          from {
-            robust
-          }
-          nonce
-          height
-          method
-          params
-          value
-        }
-      }
-    `
-    const variables = {
-      address,
-      limit: 100,
-      offset: 0
-    }
-    /** @type {{messages: GQLMessage[] | null}} */
-    const { messages } = await request(this.url, query, variables)
+    const url = `https://filfox.info/api/v1/address/${address}/messages?pageSize=100`
```

Comment:
IIUC, you are always fetching only the first 100 messages. Could this become a problem? E.g. if our wallet shows ALL transactions, we will need to load more messages beyond the first 100 progressively. (E.g. add a "load more" button in the UI that will fetch the subsequent 100 messages.)

The hard-coded limit of 100 messages is acceptable for now, but I would consider opening a follow-up GH issue if we think this is something to improve in the future.

It would also be good to show some sort of a visual indicator to let the user know when the list of transactions was truncated. I think this has a relatively higher priority.

Reply:
Fix in https://github.com/filecoin-station/filecoin-station/commit/23c1d9cc8cd7e1f40938500e60dfc1582817692f 😁

Reply:
> Fix in [23c1d9c](https://github.com/filecoin-station/filecoin-station/commit/23c1d9cc8cd7e1f40938500e60dfc1582817692f) 😁

LOL 🤣